### ISSUE TITLE
Foundation for rollup

### ISSUE THREAD

**Issue by tvvignesh** [rollup/rollup#3793]
Hi. This is not an issue but rather a proposal to rollup ecosystem.

The thing is rollup is awesome with everything it has and it is great to get started and use it and all the plugins are great in their own way but there is one thing which I see a lot.

Some good Rollup plugins from the community are not actively maintained which is understandable since everyone has their own priorities, they work for free and they have no incentive to continue working on the plugin they created. Same happens with other libs like Webpack as well.

The problem is that this may affect developers over a long run and they may even have to fork their own versions of it and maintain.

An example (there are lots like these) would be this (3 versions of the same plugin all of which are not actively maintained which is completely understandable):

https://github.com/kamilkisiela/rollup-plugin-graphql

https://github.com/bennypowers/rollup-plugin-graphql

https://github.com/Kocal/rollup-plugin-graphql

My proposal is to have a sort of a Foundation backing Rollup and its plugins (similar to Linux Foundation or ASF) which can provide governance, support and a common platform which can incentivize people to contribute plugins to the Rollup foundation. We see this happening with CNCF with lot of amazing projects under one umbrella (https://www.cncf.io/projects/) and a similar structure can probably help rollup as well.

While you may create and support official plugins but there is only so much you can support and may not be scalable over a long run.

Now, you may ask, why does a simple build tool need this and I would argue that rollup has become sort of a build platform like webpack and parcel but people can drive actual value out of rollup over a long run only if the plugins are well maintained and have a long term vision.

Feel free to close this issue if it does not seem relevant to your vision. And thanks for creating rollup 🎉

**PS:** I am very happy with rollup and I will continue to use it. But thought I should share this considering the long term vision of the project.

**Comment by shellscape** [comment]
Thanks for taking the time to write that down and the enthusiasm. 

> An example (there are lots like these) would be this (3 versions of the same plugin all of which are not actively maintained which is completely understandable):

I'd encourage you to advocate to the original author of the most stable fork to have their plugin moved to the official plugins repo, and then propose that move in the official plugins repo. @Kocal is also a maintainer in our plugins repo (last I checked) so that may work out. 

> which can provide governance

I genuinely hope that Rollup doesn't ever have the need for _governance_.

> support

We have an Open Collective for collecting donations. We could use advocates that would be willing to donate their time to help fund the project more. But I don't think we need a Foundation to make that happen.

>  While you may create and support official plugins but there is only so much you can support and may not be scalable over a long run.

Again, I don't think we need a foundation or governance for this. We have ten times the number of active plugin contributors and maintainers working on the official plugins repo than we ever did when they were separate projects. Long neglected plugins have been brought back to life thanks to these folks. Relying on community members to step up has worked out well. Working on a bundler just isn't a huge draw. After spending over a year on the Webpack team I can tell you the same holds true there as well, their userbase is just larger. 

> if the plugins are well maintained and have a long term vision.

I'd argue that we're doing a pretty good job at maintaining the plugins. And we're slowly moving widely used third party plugins into the official repo where they get more attention. As for long-term vision, the plugins which require attention to keep current are receiving that kind of attention. The commonjs and node-resolve plugins being the big two core-critical plugins. They're the two most frequently updated plugins and they tend to stay in step with the latest advancements thanks in large part to @guybedford involvement and being on the boards that discuss RFCs and where things are heading.

I'm not sure what long-term vision you're referencing or where you saw something proposed, but I think the momentum and direction has been incredibly solid thus far, and wouldn't want to see elements introduced which could disrupt that.

--

Again, I appreciate the enthusiasm here. What we could really, really use are evangelists and advocates for Rollup - two roles that we have no one helping out on. If our funding increases, we can spend more time on Rollup and less time on work-work.

I'm personally pretty biased and resistant against introducing bureaucracy where it isn't needed. We've got a great group here and we all share a maximum amount of respect, trust, and admiration for one another and I've never worked on a project so enjoyable in 20 years of being in the industry.

**Comment by tvvignesh** [comment]
@shellscape Thanks for taking the time to write this elaborate reply and I completely understand what you say.

My intention for advocating a foundation was not for the need of bueraeucracy but for the need of financial and other kind of support along with a process in place like release cycles which might help the plugins to be more sustainable. But yup I get you.

Regarding long term vision, it was not regarding rollup itself but the plugins from the community. Since each of them are individually maintained, not all of them do have a milestone or support system in place which might be a hindrance to adopting that over a long run.

The problem is that the Open Collective is for the rollup and the plugins maintained by rollup itself and not for the third party plugins. So they still work for free without any support sadly.

I definitely agree that the official plugins are actively maintained. Again, I was referring to the community driven plugins.

And I am completely on the same page with you in relation to bureaucracy. This amazing project does not need bureaucracy but a thriving ecosystem and my suggestion was in line with making the ecosystem more sustainable as much as the platform itself 🙂

Regarding evangelists, don't worry. The project speaks for itself and I will also be sharing my journey with the community on rollup in 2 months (i.e. after my startup launch) so I can help as well 🙂

To summarize, I have no worries about rollup and the official plugins. This was completely in relation with the community maintained plugins.

Feel free to close the issue. Thanks again for the time taken to reply.

**Comment by shellscape** [comment]
I'd actually vote to leave the issue open for visibility.

Third party plugins are a concern, but our doors are wide open for considering their inclusion here. I've found that, during speaking with third party authors about moving to our monorepo, they prefer to keep their projects close to the vest, even if that means not having project support. Some of it is ego (not the bad kind), some of it is pride, some of it a matter of control (again, not a bad thing), and some of it is just contentment with where their plugin is and the time they spend on it.

We've seen some major strides in community involvement and adoption over the last year, and that's a trend that I believe we'll continue to see.

**Comment by Kocal** [comment]
Hey :wave: 

Near one year ago I've already proposed to move https://github.com/Kocal/rollup-plugin-graphql under https://github.com/rollup/plugins repository, see https://github.com/rollup/plugins/issues/2#issuecomment-557902616

I don't know if newer forks exists for this plugin, since it supports Rollup 1 and multiples exports (`import { FooQuery, FooMutation } from './foo.graphql`), but I would be glad to deprecate my fork and create an official Rollup plugin.

What do you think?

**Comment by shellscape** [comment]
I say go for it

**Comment by Kocal** [comment]
PR opened at https://github.com/rollup/plugins/pull/585

**Comment by shellscape** [comment]
The graphql plugin is merged and published! Closing this issue now as resolved.

**Comment by Kocal** [comment]
Thanks!

https://github.com/Kocal/rollup-plugin-graphql has been deprecated + npm-archived, I've also opened a PR on the initial plugin to warn them about the deprecation (https://github.com/kamilkisiela/rollup-plugin-graphql/pull/10 + https://github.com/kamilkisiela/rollup-plugin-graphql/pull/7#issuecomment-717086690)
